File: backend/SpringBoot2-Java/src/test/java/com/weweb/BaseTest.java
package com.weweb;

import java.io.File;
import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.Locale;
import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.rules.TestName;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public abstract class BaseTest {

    protected static final DecimalFormat TIME_FORMAT = new DecimalFormat("####,###.###",
            new DecimalFormatSymbols(Locale.US));

    private static final String TEMP_DIR = "./target/junit-temp/" + System.currentTimeMillis();
    private final Logger logger = LoggerFactory.getLogger(getClass());

    @Rule
    public TestName name = new TestName();

    private long testStartDate;

    @Before
    public final void setUpBeforeTest() {
        testStartDate = System.currentTimeMillis();
        logger.info("===================================================================");
        logger.info("BEGIN TEST " + name.getMethodName());
        logger.info("===================================================================");

    }

    @After
    public final void tearDownAfterTest() {
        final long executionTime = System.currentTimeMillis() - testStartDate;
        logger.info("===================================================================");
        logger.info("END TEST " + name.getMethodName());
        logger.info("execution time: " + TIME_FORMAT.format(executionTime) + " ms");
        logger.info("===================================================================");
    }

    protected Logger getLogger() {
        return logger;
    }

    protected String getTempDirectory() {
        new File(TEMP_DIR).mkdirs();
        return TEMP_DIR;
    }

}

File: backend/Vertx3-Kotlin/src/test/kotlin/com/weweb/BaseTest.kt
package com.weweb

import io.vertx.core.logging.Logger
import io.vertx.core.logging.LoggerFactory
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.TestInfo
import java.io.File
import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.*

abstract class BaseTest {
    private val logger = LoggerFactory.getLogger(this.javaClass)

    private var testStartDate: Long = 0

    protected val tempDirectory: String
        get() {
            File(TEMP_DIR).mkdirs()
            return TEMP_DIR
        }

    @BeforeEach
    fun setUpBeforeTest(testInfo: TestInfo) {
        testStartDate = System.currentTimeMillis()
        logger.info("===================================================================")
        logger.info("BEGIN TEST " + testInfo.displayName)
        logger.info("===================================================================")

    }

    @AfterEach
    fun tearDownAfterTest(testInfo: TestInfo) {
        val executionTime = System.currentTimeMillis() - testStartDate
        logger.info("===================================================================")
        logger.info("END TEST " + testInfo.displayName)
        logger.info("execution time: " + TIME_FORMAT.format(executionTime) + " ms")
        logger.info("===================================================================")
    }

    protected fun logger(): Logger {
        return logger
    }

    companion object {

        protected val TIME_FORMAT = DecimalFormat("####,###.###",
                DecimalFormatSymbols(Locale.US))

        private val TEMP_DIR = "./target/junit-temp/" + System.currentTimeMillis()
    }

}
